shell/app-shell/app-shell.js

```diff
@@ -217,6 +223,12 @@ class AppShell extends Xen.Base {
       key
     });
   }
+  _calculateShareState(config, key, user) {
+    // calculate sharing state
+    let isProfile = user.profiles && user.profiles[key];
+    let isShared = user.shares && user.shares[key];
+    return isShared ? Const.SHARE.friends : isProfile ? Const.SHARE.self : Const.SHARE.private;
```

Comment:
This feels fundamental so an expanded version (chained `if` or a `switch`) might be clearer, and a unit test might help out as well.

Reply:
That being said the logic looks sound to me.

Reply:
The sharing thing needs TLC in general. I left it scraggly because I just needed it to work and I didn't want to do a half-job. I'll add a ticket for an overhaul of the sharing logic .